packages/marketplace/test/Exchange.test.ts

```diff
@@ -19,8 +19,29 @@ import {
   isOrderEqual,
 } from './utils/order.ts';
 import {ZeroAddress, AbiCoder} from 'ethers';
+import {upgrades} from 'hardhat';
 
 describe('Exchange.sol', function () {
+  it('should upgrade the contract successfully', async function () {
+    const {ExchangeContractAsDeployer, ExchangeUpgradeMock} = await loadFixture(
+      deployFixtures
+    );
+    const protocolFeePrimary =
+      await ExchangeContractAsDeployer.protocolFeePrimary();
+    const protocolFeeSec =
+      await ExchangeContractAsDeployer.protocolFeeSecondary();
+    const feeReceiver = await ExchangeContractAsDeployer.defaultFeeReceiver();
+
```

Comment:
Can you add this validation in the tests ?
https://docs.openzeppelin.com/upgrades-plugins/1.x/api-hardhat-upgrades#validate-upgrade